src/MassTransit.Abstractions/IMetricsContext.cs

```diff
@@ -0,0 +1,15 @@
+﻿namespace MassTransit
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Provides access to tags added to metrics.
+    /// </summary>
+    public interface IMetricsContext
+    {
+        /// <summary>
+        /// Gets the tag collection.
+        /// </summary>
+        ICollection<KeyValuePair<string, object?>> Tags { get; }
```

Comment:
`IReadOnlyCollection` or `IEnumerable` to avoid modification?

Reply:
This is a SET, not a GET, so I'd rather have a method called Add, vs exposing the collection, and using the underlying tags type as mentioned so it follows the same semantics.